### DISCUSSION TITLE
Pull request to a repository runs the workflow without access to secrets.

### DISCUSSION THREAD

**Discussion by elongl** [community/community#33683] [Actions]
I have an open-source repository.
A contributor opened a PR to the repository which runs a GitHub Action that tests his changes.
The action ran without the secrets' context, meaning that they all evaluated to empty strings and therefore didn't run the tests properly. I don't want to merge the pull request before the tests pass.
How could I allow a pull request to have the secrets context so that it can test against my environments?
Thanks.

**Comment by airtower-luna** [comment]
In short, you don't. Otherwise anyone could edit the workflow to extract your secrets, create a PR, and do whatever they want with your secrets.

There are ways to run workflows that are triggered by pull requests in other ways with access to secrets, but you have to be very careful not to introduce ways that allow people to steal your secrets. See: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Reply by elongl** [reply]
I expect GitHub to always run the workflow in `main` rather than the one in the PR though.
Therefore, if I trust my workflow to not leak the secrets, there shouldn't be a problem.

Thanks for sharing the link, I'll look into it.

**Reply by airtower-luna** [reply]
> I expect GitHub to always run the workflow in main rather than the one in the PR though.

That depends on the exact event you're looking for. A `pull_request` event runs on a temporary merge commit.
